packages/teleport/src/DesktopSession/DesktopSession.tsx

```diff
@@ -40,42 +46,117 @@ export function DesktopSession(props: State) {
-
   const clipboardProcessing =
     clipboardState.enabled && clipboardState.permission.state === 'prompt';
 
-  // Websocket is closed but we haven't
-  // closed it on purpose or registered a tdp error.
-  const unknownConnectionError =
-    wsConnection === 'closed' &&
-    !disconnected &&
-    tdpConnection.status === 'success';
-
   const processing =
     fetchAttempt.status === 'processing' ||
     tdpConnection.status === 'processing' ||
     clipboardProcessing;
 
-  let alertText: string;
-  if (fetchAttempt.status === 'failed') {
-    alertText = fetchAttempt.statusText || 'fetch attempt failed';
-  } else if (tdpConnection.status === 'failed') {
-    alertText = tdpConnection.statusText || 'tdp connection failed';
-  } else if (clipboardError) {
-    alertText = clipboardState.errorText || 'clipboard sharing failed';
-  } else if (unknownConnectionError) {
-    alertText = 'Session disconnected for an unknown reason';
-  }
+  // Manages the state of the error dialog.
+  const [errorDialog, setErrorDialog] = useState({
+    open: false,
+    text: '',
+    fatal: false,
+  });
+
+  // onDialogClose is called when a user
+  // dismisses a non-fatal error dialog.
+  const onDialogClose = () => {
+    // This setTdpConnection call will cause the useEffect below
+    // to calculate the errorDialog state.
+    setTdpConnection(prevState => {
+      // onDialogClose should only be called when
+      // the user dismisses a non-fatal error dialog,
+      // and prevState.status === '' means non-fatal
+      // error dialog, so the below if statement should
+      // always be true.
+      if (prevState.status === '') {
+        // If prevState.status was a non-fatal error,
+        // we assume that the TDP connection remains open.
+        return { status: 'success' };
+      }
+      return prevState;
+    });
+  };
 
-  if (alertText) {
+  // Calculate the state of the error dialog based on the various
+  // sub-states which determine it.
+  useEffect(() => {
+    const clipboardError = clipboardState.enabled && clipboardState.errorText;
+
+    // Websocket is closed but we haven't
+    // closed it on purpose or registered a fatal tdp error.
+    const unknownConnectionError =
+      wsConnection === 'closed' &&
+      !disconnected &&
+      (tdpConnection.status === 'success' || tdpConnection.status === '');
+
+    let errorText = '';
+    if (fetchAttempt.status === 'failed') {
+      errorText = fetchAttempt.statusText || 'fetch attempt failed';
+    } else if (tdpConnection.status === 'failed') {
+      errorText = tdpConnection.statusText || 'tdp connection failed';
+    } else if (tdpConnection.status === '') {
+      errorText = tdpConnection.statusText || 'encountered a non-fatal error';
+    } else if (clipboardError) {
+      errorText = clipboardState.errorText || 'clipboard sharing failed';
+    } else if (unknownConnectionError) {
+      errorText = 'Session disconnected for an unknown reason';
```

Comment:
```suggestion
      errorText = 'Session disconnected for an unknown reason. Try reloading (or refreshing) the browser.';
```

and then the button could say `Reload` or `Refresh`? 

also i'm confused whether our error messages should be all lower cap or properly capitalized with punctuation. I'm just seeing a mixture of both. It's not a big deal to me but just wondering in general

Reply:
> also i'm confused whether our error messages should be all lower cap or properly capitalized with punctuation.

In my opinion if we're displaying it as a message to the user, we should strive to give it proper capitalization and punctuation.